src/target/stream.c

```diff
@@ -85,6 +118,25 @@ stumpless_open_stream_target( const char *name, FILE *stream ) {
   return NULL;
 }
 
+void
+stumpless_set_severity_color( struct stumpless_target *target, enum stumpless_severity severity, const char *escape_code )
+{
+  if (severity > 7)
+    raise_index_out_of_bounds("Severity value must be part of 'stumpless_severity' enum (0-7)", severity);
+
+  if (target->type != STUMPLESS_STREAM_TARGET)
+    raise_target_unsupported("This function is only supported for stream targets");
+
+  struct stream_target *starget = (struct stream_target *)target->id;
+  lock_target(target);
```

Comment:
acquire this lock before checking the type, so that it is locked for the duration of the code that uses it

Reply:
Done.